File: src/test/java/de/thelooter/eventchecker/EventCheckerTest.java
package de.thelooter.eventchecker;

import be.seeseemelk.mockbukkit.MockBukkit;
import be.seeseemelk.mockbukkit.ServerMock;
import de.thelooter.eventchecker.EventChecker;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertInstanceOf;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

class EventCheckerTest {

  ServerMock server;
  EventChecker eventChecker;

  @BeforeEach
  void setUp() {
    server = MockBukkit.mock();
    eventChecker = MockBukkit.load(EventChecker.class);
  }

  @AfterEach
  void tearDown() {
    MockBukkit.unmock();
  }

  @Test
  void test() {
    assertNotNull(eventChecker);
    assertTrue(eventChecker.isEnabled());
    assertInstanceOf(EventChecker.class, eventChecker);
  }
}

File: src/test/kotlin/de/thelooter/eventchecker/EventCheckerTest.kt
package de.thelooter.eventchecker

import org.mockbukkit.mockbukkit.MockBukkit
import org.mockbukkit.mockbukkit.ServerMock
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class EventCheckerTest {
    private var server: ServerMock? = null
    private var eventChecker: EventChecker? = null

    @BeforeEach
    fun setUp() {
        server = MockBukkit.mock()
        eventChecker = MockBukkit.load(EventChecker::class.java)
    }

    @AfterEach
    fun tearDown() {
        MockBukkit.unmock()
    }

    @Test
    fun test() {
        Assertions.assertNotNull(eventChecker)
        Assertions.assertTrue(eventChecker!!.isEnabled)
        Assertions.assertInstanceOf(EventChecker::class.java, eventChecker)
    }
}